src/components/ModalConfirmation/ModalConfirmation.tsx

```diff
@@ -3,39 +3,26 @@ import classnames from 'classnames';
 import styles from './styles.module.scss';
 import confirmationPic from './chinatown-logo.png';
 
-interface Props {
-  handleClose: (event: React.MouseEvent<HTMLButtonElement>) => void;
-  showConfirmModal: boolean;
-}
+import { useModalPaymentDispatch } from "../../utilities/hooks/ModalPaymentContext/context"
+import { CLOSE_MODAL } from "../../utilities/hooks/ModalPaymentContext/constants"
 
-const ModalConfirmation: React.SFC<Props> = ({
-  handleClose,
-  showConfirmModal,
-}) => {
-  return (
-    <form
-      id="payment-form"
-      className={classnames(styles.container, 'modalForm--form')}
-      style={{ display: showConfirmModal ? 'block' : 'none' }}
-    >
-      <button className={'closeButton--close'} onClick={handleClose}>
-        {' '}
-        ×
-      </button>
+const ModalConfirmation: React.SFC = () => {
+  const dispatch = useModalPaymentDispatch();
 
+  return (
+    <div className={styles.paymentContainer}>
       <h2>Thank you for your donation!</h2>
       <p>We appreciate your support for small businesses</p>
 
       <img className={styles.image} src={confirmationPic} alt="Logo" />
 
       <button
         className={classnames(styles.finishBtn, 'modalButton--filled')}
-        onClick={handleClose}
+        onClick={() => dispatch({type: CLOSE_MODAL, payload: undefined})}
```

Comment:
shouldn't need to send a payload of undefined?

Reply:
i made it so that all actions require a payload, undefined dispatches have a specific action in changing the the state that don't require a payload